src/librbd/ImageCtx.cc

```diff
@@ -596,7 +596,7 @@ struct C_InvalidateCache : public Context {
   bool ImageCtx::test_features(uint64_t in_features,
                                const RWLock &in_snap_lock) const
   {
-    assert(snap_lock.is_locked());
+    assert(in_snap_lock.is_locked());
```

Comment:
The convention of passing in a reference of a lock is just a way to indicate that this method requires that you have already locked the lock.  The assertion just verifies that fact. With this change, I could now invoke ```test_features(XYZ, owner_lock)``` and the assert will pass if I have ```ImageCtx::owner_lock``` locked without locking the correct ```ImageCtx::snap_lock```.